gtk4/Cargo.toml

```diff
@@ -30,6 +30,7 @@ cairo-rs = {git = "https://github.com/gtk-rs/gtk-rs-core", version = "0.16"}
 ffi = {package = "gtk4-sys", path = "./sys", version = "0.5.0"}
 field-offset = "0.3"
 futures-channel = "0.3"
+futures-executor = "0.3"
```

Comment:
Where is that actually used?

Reply:
It's used to block on a channel:
https://github.com/gtk-rs/gtk4-rs/blob/fa3999913742fd38f2284033a445223a13edb0f1/gtk4/src/lib.rs#L79
I guess this could be rewritten to emulate it using a `glib::MainLoop` and `idle_add`?

Reply:
You could create a new main context and `block_on()` with it, that would be equivalent. Or you use a `std::sync::mpsc` channel and just block on the receiver directly without all the async complications.

Reply:
Changed it to `std::sync::mpsc`